compiler/src/dotty/tools/dotc/transform/patmat/Space.scala

```diff
@@ -815,16 +817,15 @@ class SpaceEngine(using Context) extends SpaceLogic {
         isCheckable(and.tp1) || isCheckable(and.tp2)
       }) ||
       tpw.isRef(defn.BooleanClass) ||
-      classSym.isAllOf(JavaEnumTrait)
+      classSym.isAllOf(JavaEnumTrait) ||
+      classSym.is(Case) && {
+        if seen.add(tpw) then productSelectorTypes(tpw, sel.srcPos).exists(isCheckable(_))
+        else true // recursive case class: return true and other members can still fail the check
```

Comment:
It seems `false` aligns better with the protocol that if a component is `sealed`, then perform the check.

Reply:
Given something like `case class Foo(parent: Option[Foo], other: Bar)` this else branch is hit when we've running `isCheckable(Foo)` on the Foo in the Option.  My thinking is we can optimistically return `true` here, and then whether or not `Bar` is checkable will determine if the top `Foo` is checkable.  So I think `true` is right here, no?

Reply:
Given the following type:

```Scala
case class A(a: A)
```
IIRC, the implementation above will have `isCheckable(A) == true`. If that's as expected, then `true` is fine. Otherwise, it seems `false`  should be used here.

Reply:
Yeah, I'm happy with that result.